File: VB-Examples/16_ImageAndShape/AddShapes.vb
Imports Spire.Doc
Imports Spire.Doc.Documents
Imports Spire.Doc.Fields

Namespace AddShapes
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			'Create Word document.
			Dim doc As New Document()

			'Add a section to the document
			Dim sec As Section = doc.AddSection()

			'Add a paragraph to the section
			Dim para As Paragraph = sec.AddParagraph()
			Dim x As Integer = 60, y As Integer = 40, lineCount As Integer = 0
			For i As Integer = 1 To 19

				'Check if the current line count is a multiple of 8
				If lineCount > 0 AndAlso lineCount Mod 8 = 0 Then

					'Append a page break to start a new page
					para.AppendBreak(BreakType.PageBreak)
					x = 60
					y = 40
					lineCount = 0
				End If

				'Append a shape to the paragraph
				Dim shape As ShapeObject = para.AppendShape(50, 50, CType(i, ShapeType))
				shape.HorizontalOrigin = HorizontalOrigin.Page
				shape.HorizontalPosition = x
				shape.VerticalOrigin = VerticalOrigin.Page
				shape.VerticalPosition = y + 50
				x = x + CInt(shape.Width) + 50

				'Check if the shape count is a multiple of 5
				If i > 0 AndAlso i Mod 5 = 0 Then

					'Adjust the vertical position and line count
					y = y + CInt(shape.Height) + 120
					lineCount += 1
					x = 60
				End If

			Next i

			'Save the document
			doc.SaveToFile("AddShape.docx", FileFormat.Docx)

			'Dispose the document
			doc.Dispose()

			'Launch Word file.
			WordDocViewer("AddShape.docx")
		End Sub

		Private Sub WordDocViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace

File: CS-Examples/16_ImageAndShape/AddShapes.cs
using System;
using System.Windows.Forms;
using Spire.Doc;
using Spire.Doc.Documents;
using Spire.Doc.Fields;

namespace AddShapes
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Create a new document
			Document doc = new Document();

			// Add a section to the document
			Section sec = doc.AddSection();

			// Add a paragraph to the section
			Paragraph para = sec.AddParagraph();

			int x = 60, y = 40, lineCount = 0;
			for (int i = 1; i < 20; i++)
			{
				// Check if the current line count is a multiple of 8
				if (lineCount > 0 && lineCount % 8 == 0)
				{
					// Append a page break to start a new page
					para.AppendBreak(BreakType.PageBreak);
					x = 60;
					y = 40;
					lineCount = 0;
				}

				// Append a shape to the paragraph
				ShapeObject shape = para.AppendShape(50, 50, (ShapeType)i);
				shape.HorizontalOrigin = HorizontalOrigin.Page;
				shape.HorizontalPosition = x;
				shape.VerticalOrigin = VerticalOrigin.Page;
				shape.VerticalPosition = y + 50;
				x = x + (int)shape.Width + 50;

				// Check if the shape count is a multiple of 5
				if (i > 0 && i % 5 == 0)
				{
					// Adjust the vertical position and line count
					y = y + (int)shape.Height + 120;
					lineCount++;
					x = 60;
				}
			}

			// Save the document
			doc.SaveToFile("AddShape.docx", FileFormat.Docx);

			// Dispose the document
			doc.Dispose();

            //Launch Word file.
            WordDocViewer("AddShape.docx");
        }

        private void WordDocViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}